tests/10.ABI.ts

```diff
@@ -218,214 +219,228 @@ describe('ABI type checking', () => {
+    newTestCase(new ABIBoolType(), true, new Uint8Array([128])),
+    newTestCase(new ABIBoolType(), false, new Uint8Array([0])),
+    newTestCase(
+      new ABIStringType(),
+      'asdf',
+      new Uint8Array([0, 4, 97, 115, 100, 102])
+    ),
+    newTestCase(
+      new ABIArrayStaticType(new ABIBoolType(), 3),
+      [true, true, false],
+      new Uint8Array([192])
+    ),
+    newTestCase(
+      new ABIArrayStaticType(new ABIBoolType(), 8),
+      [false, true, false, false, false, false, false, false],
+      new Uint8Array([64])
+    ),
+    newTestCase(
+      new ABIArrayStaticType(new ABIBoolType(), 8),
+      [true, true, true, true, true, true, true, true],
+      new Uint8Array([255])
+    ),
+    newTestCase(
+      new ABIArrayStaticType(new ABIBoolType(), 9),
+      [true, false, false, true, false, false, true, false, true],
+      new Uint8Array([146, 128])
+    ),
+    newTestCase(
+      new ABIArrayStaticType(new ABIUintType(64), 3),
+      [BigInt(1), BigInt(2), 3], // Deliberately mix BigInt and int
+      new Uint8Array([
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        1,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        2,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        3,
+      ])
+    ),
+    newTestCase(
+      new ABIArrayDynamicType(new ABIBoolType()),
+      [],
+      new Uint8Array([0, 0])
+    ),
+    newTestCase(
+      new ABIArrayDynamicType(new ABIBoolType()),
+      [true, true, false],
+      new Uint8Array([0, 3, 192])
+    ),
+    newTestCase(
+      new ABIArrayDynamicType(new ABIBoolType()),
+      [false, true, false, false, false, false, false, false],
+      new Uint8Array([0, 8, 64])
+    ),
+    newTestCase(
+      new ABIArrayDynamicType(new ABIBoolType()),
+      [true, false, false, true, false, false, true, false, true],
+      new Uint8Array([0, 9, 146, 128])
+    ),
+    newTestCase(ABIType.from('()'), [], new Uint8Array([])),
+    // 2^6 + 2^5 = 64 + 32 = 96
+    newTestCase(
+      ABIType.from('(bool,bool,bool)'),
+      [false, true, true],
+      new Uint8Array([96])
+    ),
+    newTestCase(
+      ABIType.from('(bool[3])'),
+      [[false, true, true]],
+      new Uint8Array([96])
+    ),
+    newTestCase(
+      ABIType.from('(bool[])'),
+      [[false, true, true]],
+      new Uint8Array([0, 2, 0, 3, 96])
+    ),
+    newTestCase(
+      ABIType.from('(bool[2],bool[])'),
+      [
+        [true, true],
+        [true, true],
+      ],
+      new Uint8Array([192, 0, 3, 0, 2, 192])
+    ),
+    newTestCase(
+      ABIType.from('(bool[],bool[])'),
+      [[], []],
+      new Uint8Array([0, 4, 0, 6, 0, 0, 0, 0])
+    ),
+    newTestCase(
+      ABIType.from('(string,bool,bool,bool,bool,string)'),
+      ['AB', true, false, true, false, 'DE'],
+      new Uint8Array([0, 5, 160, 0, 9, 0, 2, 65, 66, 0, 2, 68, 69])
+    ),
+  ].forEach((testCase) => {
```

Comment:
Could we add a testcase to test a number inside a tuple, e.g. `(uint8)`.

Reply:
@algochoi Good call - I added a test in https://github.com/algorand/js-algorand-sdk/pull/701/commits/cc4e928fda13e1b60d9aa10fe42d171984118bf0.  Can you resolve if it looks good?

Aside - Passing remarks:
* While adding the test, I discovered a bug in the test implementation.  I had inadvertently missed the return statement:  https://github.com/algorand/js-algorand-sdk/pull/701/commits/cc4e928fda13e1b60d9aa10fe42d171984118bf0#diff-002803be91ba5405e645aec5c397e490594a1883499139d9c69c070a594fd9dcR422.  I wish TypeScript treated if-statements as an expression.  Returning a value (like Scala) prevents this class of problem.
* I just realized the pre-commit hook will transform code.  I wrote _else if_ statements that got moved to if-statements.

Reply:
Thanks, I think this looks good - resolving